### PR TITLE
feat: Java workshop

### PR DESCRIPTION
**Pull request by ybezsonov**
# What problem does it solve?

Summarize the changes and the motivation behind your changes. 

- Add content for Java version for EKS for Developers. It has Introduction/Containers/EKS and additional Java specific Optimizations section.

## Type of change

- [ ] Bug fix or improvement (non-breaking change which fixes an issue)
- [ ] New lab exercise (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing workshop materials to not work as expected)

# How Can Other Contributors Test Your Changes?

Describe the tests that you ran to verify your changes so that other contributors can reproduce your steps. 

- [ ] N/A
- [ ] Describe below:

# Checklist:

- [x] My contributions follow the [style guidelines](../docs/style-guide.md) of this project
- [x] I have performed a self-review of my changes
- [ ] I have added tests that prove my changes are effective

### CODE REVIEW

File: website/docs/optimizations/java/optimized-jvm.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+ENV OTEL_IMR_EXPORT_INTERVAL "10000"
+ENV OTEL_EXPORTER_OTLP_ENDPOINT "http://localhost:4317"
+
+EXPOSE 8080
+ENTRYPOINT ["./custom-jre/bin/java","-jar","-Dserver.port=8080","/store-spring.jar"]
+```
+
+As discussed in the first chapter the Dockerfile leverages a multi-stage build. In the first stage (starting line 1), the application and a custom runtime is built. The custom runtime is based on the list of dependencies that have been collected with `jdeps` (see line 11 - 14).
+
+In the first step, the complete classpath of the application is analyzed and all module dependencies are written in a file called `jre-deps.info`. You may have noticed the explicit addition of `jdk.crypto.ec`. This module provides the implementation of the SunEC security provider which is necessary for TLS support. Unfortunately, it is not obvious from the analysis of the classpath that this additional module is necessary. For this reason, it is added at this point.
```
Comment (tucktuck9) [inline website/docs/optimizations/java/optimized-jvm.md:109]: In the first step, we analyze the entire classpath of the application and write down all module dependencies in a file called `jre-deps.info`. You might have noticed that we explicitly added `jdk.crypto.ec`. This module contains the implementation of the SunEC security provider, which is essential for TLS support. However, it is not easy to determine from the classpath analysis that this module is needed, which is why we include it at this stage.

File: website/docs/optimizations/java/optimized-jvm.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+The `jre-deps.info` is used as input for `jlink` (line 20 - 23) in order to build a custom runtime. The goal of this is to reduce the size as much as possible, that's the reason the runtime is compressed and no header files and no man-pages are included.
+
+In the second stage of our build (line 28 - 29), we copy the JAR file of the application as well the custom runtime to the target image.
+
+## 3. Re-deploying the application
+
+After the new image has been pushed to ECR you can re-trigger the deployment of the application:
```
Comment (tucktuck9) [inline website/docs/optimizations/java/optimized-jvm.md:117]: After pushing the new image to ECR, you can re-trigger the deployment of the application:

File: website/docs/optimizations/java/optimized-jvm.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+```bash showLineNumbers
+kubectl logs $(kubectl get pods -n unicorn-store-spring -o json | jq --raw-output '.items[0].metadata.name') -n unicorn-store-spring
+kubectl logs $(kubectl get pods -n unicorn-store-spring -o json | jq --raw-output '.items[0].metadata.name') -n unicorn-store-spring | grep "Started StoreApplication"
+```
+
+![optimized-jvm-eks](./images/optimized-jvm-eks.png)
+
+## Conclusion
+
+As you can see, we were able to reduce the size of the container image from 380 MB to 257 MB. The application startup time improved slightly.
```
Comment (tucktuck9) [inline website/docs/optimizations/java/optimized-jvm.md:154]: As you can see, we managed to decrease the size of the container image from 380 MB to 257 MB. The application startup time also improved by a small margin.

File: website/docs/optimizations/java/results.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+---
+title: "Results of the optimization steps"
+sidebar_position: 6
+---
+
+For the different labs, we looked at one optimization at a time: we started with the initial container image without optimization, used jlink and jdeps for a custom runtime, introduced jib, and finally used GraalVM. The following table shows the different versions with the image size and the start time of the application (tested using Amazon EKS).
```
Comment (tucktuck9) [inline website/docs/optimizations/java/results.md:6]: In previous labs, we looked at one optimization at a time. We started with the initial container image without optimization, then used 'jlink' and 'jdeps' for a custom runtime, introduced 'jib', and finally used GraalVM. The following table shows the different versions with the image size and the start time of the application (tested using Amazon EKS).

File: website/docs/optimizations/java/results.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+| Version         | Image Size | Start time (p99) |
+| -----------     |------------|------------------|
+| No optimization | 380MB      | 12.5s            |
+| Custom JRE      | 257MB      | 12.1s            |
+| Jib             | 212MB      | 10.5s            |
+| GraalVM         | 166MB      | 0.92s            |
+
+We can see very clearly from the table that different optimizations lead to different results. All optimizations show a significant reduction in the size of the container image. In terms of startup times, GraalVM clearly stands out with less than one second, for the well-known reasons.
```
Comment (tucktuck9) [inline website/docs/optimizations/java/results.md:15]: We can see very clearly from the table that different optimizations lead to different results. All optimizations show a significant reduction in the size of the container image. In terms of startup times, GraalVM clearly stands out with less than one second, for well-known reasons.

File: website/docs/optimizations/java/results.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+But which optimization step should customers use in their applications? On the one hand, this depends very much on what the optimization goal is and what technical skills are available in the teams. The simplest of the optimization techniques mentioned is the use of Jib, as this hardly requires any changes in the build step. For a custom runtime, extensive changes in the Dockerfile are necessary. In addition, jdeps can only detect compile time dependencies, any runtime dependencies have to be added manually. The most difficult optimization is to use native-image and GraalVM, especially for applications that have been implemented some time ago without GraalVM in mind.
```
Comment (tucktuck9) [inline website/docs/optimizations/java/results.md:17]: The question remains, which optimization step should you use with your applications? On the one hand, this depends very much on what your optimization goal and technical skills available. The simplest of the optimization techniques is the use of Jib, as this hardly requires any changes in the build step. For a custom runtime, extensive changes in the Dockerfile are needed. In addition, 'jdeps' can only detect compile time dependencies, and any runtime dependencies have to be added manually. The most difficult optimization is using 'native-image' and 'GraalVM', especially for applications that have been implemented some time ago without GraalVM in mind.

File: website/docs/optimizations/java/summary.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
+---
+title: "Summary"
+sidebar_position: 1000
+---
+
+During this workshop you have built container images with Spring Boot Java Application, applied different optimization technics to container images and deployed container images with Java Application to Amazon EKS cluster.
```
Comment (tucktuck9) [inline website/docs/optimizations/java/summary.md:6]: During this workshop, you've learned how to create container images using the Spring Boot Java Application. You've also explored various optimization techniques for these container images and successfully deployed them, along with your Java Application, to an Amazon EKS cluster.

**Review by tucktuck9 — COMMENTED** [review]
Docs-only review with minor grammatical and flow improvements

**Comment by ybezsonov** [comment]
@tucktuck9 Thank you for the great review. I applied all the comments.

**Review by tucktuck9 — APPROVED** [review]
LGTM! Let's sync up on Slack before we merge, given that makes this whole thing "live."

File: website/docs/introduction/java/about-workshop.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+sidebar_position: 1
+---
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+In this workshop, learn how to build cloud-native Java applications and deploy to Amazon Elastic Kubernetes Service (EKS) with best practices and performance-optimization techniques. Get hands-on experience using Amazon EKS, Amazon Corretto, Amazon ECR, GraalVM, Spring Boot, and more.
+
+The workshop covers a variety of AWS services and tools and provides an introduction to Java related concepts. Below you can find a broad overview of services that are covered throughout the modules:
+
+![java-on-aws-eks](./images/java-on-amazon-eks.png)
```
Comment (abencomoc) [inline website/docs/introduction/java/about-workshop.md:12]: Just a comment that can be fixed later on.
This image shows incoming requests received by ALB and sent to aws-lb-controller. Actually, aws-lb-controller is not in the network flow, ALB sends requests directly to App pod/container (or K8s svc).
Users are often confused thinking that network traffic goes through the aws-lb-controller, which is not correct.

File: website/docs/eks/java/eks-create.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+
+- [Building and Running Multi-Architecture Containers](../../containers/java/upload-ecr.md)
+
+[Amazon Elastic Kubernetes Service (Amazon EKS)](https://aws.amazon.com/eks/) is a managed service that you can use to run Kubernetes on AWS without needing to install, operate, and maintain your own Kubernetes control plane or nodes. Kubernetes is an open-source system for automating the deployment, scaling, and management of containerized applications.
+
+## 1. Running the UnicornStore with Amazon EKS
+
+Below you can find a visualization on how the UnicornStore application will be deployed and communicate with Amazon EKS:
+
+![unicornstore-architecture-eks](./images/unicornstore-architecture-eks.png)
```
Comment (abencomoc) [inline website/docs/eks/java/eks-create.md:29]: Just a comment that can be fixed later on.
This image shows incoming requests received by ALB and sent to aws-lb-controller. Actually, aws-lb-controller is not in the network flow, ALB sends requests directly to App pod/container (or K8s svc).
Users are often confused thinking that network traffic goes through the aws-lb-controller, which is not correct.

**Review by abencomoc — APPROVED** [review]
Just a suggestion to improve architecture diagram (png image). This change can be added later, no need to delay java workshop commit.
Java workshop content looks good.

**Pull request merged**